packages/react-sdk/src/components/image.tsx

```diff
@@ -1,24 +1,85 @@
 import React, { forwardRef, type ElementRef, type ComponentProps } from "react";
+import { getImageAttributes, type ImageLoader } from "../component-utils/image";
+
+const getInt = (x: unknown): number | undefined => {
+  if (typeof x === "number") {
+    return Math.round(x);
+  }
+
+  if (typeof x === "string") {
+    const v = Number.parseInt(x, 10);
+    if (`${v}` === x) {
```

Comment:
why do you have template here lol
```suggestion
    if (parsedValue === value) {
```

Reply:
So following this logic Number.parseInt("10.5", 10) === 10
and it will be false because "10" !== ''10.5"

so you will end up returning undefined, which is probably wrong, because you wouldn't be supporting floating comma string values

while above you do Math.round() which tells me that you actually do want to support floating comma.

Reply:
so to be correct, you probably need to use Number.parseFloat()

Reply:
parseInt('101.1', 10) gives 101 I'm just checking that it was right Int in x
`${101}` !== `101.1` and should work for intergers only

Reply:
I don't like js `==`comparison because its some kind of undefined behaviour (too much to remember), so convert into right type

Reply:
so you want to return undefined when user passes "10.5"?

Reply:
> I don't like js `==`comparison because its some kind of undefined behaviour (too much to remember), so convert into right type

me neither, in fact if you look into the code base, you should only find == when comparing to null, to cover both null and undefined, we never use == for anything else

Reply:
Please add an eslint rule for both

1. == (only allowed to compare to null)
2. some min length for variables min length

Reply:
Yes.  By spec it's an integer value. Having the right inputs someday we can remove all that checks.

Reply:
Then I don't understand why we don't return undefined if we pass `10.5` as a number, but instead we round it to `10`

Reply:
Fixed, really strange, thank you!